tests/Microsoft.Identity.Test.Unit/TelemetryTests/RegionalTelemetryTests.cs

```diff
@@ -138,9 +137,86 @@ public async Task TelemetrySerializedTokenCacheTestAsync()
             }
         }
 
+        /// <summary>
+        /// Acquire token for client with regionToUse when auto region discovery fails
+        ///    Current_request = 2 | ATC_ID, 0 | centralus, 1, 1,
+        ///    Last_request = 2 | 0 | | |
+        /// </summary>
+        [TestMethod]
+        public async Task TelemetryUserProvidedRegionAutoDiscoveryFailsTestsAsync()
+        {
+            Trace.WriteLine("Acquire token for client with region provided by user and region detection fails.");
+            _harness.HttpManager.AddMockHandlerContentNotFound(HttpMethod.Get, TestConstants.ImdsUrl);
+            var result = await RunAcquireTokenForClientAsync(AcquireTokenForClientOutcome.UserProvidedRegion).ConfigureAwait(false);
+            AssertCurrentTelemetry(result.HttpRequest,
+                ApiIds.AcquireTokenForClient,
+                forceRefresh: false,
+                "4",
+                isCacheSerialized: false);
+            AssertPreviousTelemetry(result.HttpRequest, expectedSilentCount: 0);
+        }
+
+        /// <summary>
+        /// Acquire token for client with regionToUse when auto region discovery passes with region same as regionToUse
+        ///    Current_request = 2 | ATC_ID, 0 | centralus, 1, 1, 1
+        ///    Last_request = 2 | 0 | | |
+        /// </summary>
+        [TestMethod]
+        public async Task TelemetryUserProvidedRegionAutoDiscoverRegionSameTestsAsync()
+        {
+            try
+            {
+                Environment.SetEnvironmentVariable(TestConstants.RegionName, TestConstants.Region);
+
+                Trace.WriteLine("Acquire token for client with region provided by user and region detected is same as regionToUse.");
+                var result = await RunAcquireTokenForClientAsync(AcquireTokenForClientOutcome.UserProvidedRegion).ConfigureAwait(false);
+                AssertCurrentTelemetry(result.HttpRequest,
+                    ApiIds.AcquireTokenForClient,
+                    forceRefresh: false,
+                    "1",
+                    isCacheSerialized: false,
+                    isvalidUserProvidedRegion: "1");
+                AssertPreviousTelemetry(result.HttpRequest, expectedSilentCount: 0);
+            }
+            finally
+            {
+                Environment.SetEnvironmentVariable(TestConstants.RegionName, "");
```

Comment:
>Environment.SetEnvironmentVariable(TestConstants.RegionName, ""); [](start = 16, length = 65)

move to test cleanup? similar to what you did in the other test class?

Reply:
Done